### DISCUSSION TITLE
New Next.js deployment breaks link prefetching for anyone currently browsing the site.

### DISCUSSION THREAD

**Discussion by ericmerl** [vercel/community#853] [Help] [answered]
I'm using next/link with prefetching. If I deploy a new build, the prefetch request tries to pull from the previous build files and returns a 404. The prefetch request is going to /_next/data/buildID/. The build ID folder doesn't exist after the new deployment. I'm pretty sure this behavior is expected.

The issue I'm having is that if you click the link, it isn't doing a full page refresh. Instead, it tries to do the client transition. The props for the client transition are empty, so it throws errors because the expected props for the page are missing.

This is only happening for users who are currently on the site when the new build is deployed. If you encounter the issue and refresh the page, the site works as expected.

Is there a way to catch this prefetch 404 and trigger a page reload? Or should that be happening already, and maybe something in my configuration is preventing it?

I'm using Next.js 12.3.0.

**Comment by amyegan** [comment]
Thanks for reaching out @ericmerl! I have a few questions to help us locate the cause and figure out a solution.
- Can you confirm that this is new behavior (not something that may have been happening in previous deployments)? 
- What version of Next.js were you using before updating to 12.3.0?
- Does it get better if you roll back to the previous version of Next.js?

Please also share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or your repo if it is public. That will help with tracing the error and debugging

**Comment by ericmerl** [comment, marked as answer]
I was previously using 12.2.0 and had the same issue, I upgraded to 12.3.0 today to see if that helped resolve anything.

I started this specific project a few months ago, and I believe it has been happening since the project started.

I'll work on putting together a reproducible example for you.

If it's helpful, I believe these are all people with a similar issue:

[https://github.com/vercel/next.js/discussions/38414](https://github.com/vercel/next.js/discussions/38414)

[https://stackoverflow.com/questions/68009879/nextjs-link-prefetch-returns-404](https://stackoverflow.com/questions/68009879/nextjs-link-prefetch-returns-404)

[https://stackoverflow.com/questions/71385135/vercel-next-js-seeminlgy-randomly-returns-404s-with-notfound-true](https://stackoverflow.com/questions/71385135/vercel-next-js-seeminlgy-randomly-returns-404s-with-notfound-true)

[https://github.com/vercel/next.js/discussions/36670?sort=top](https://github.com/vercel/next.js/discussions/36670?sort=top)

**Reply by amyegan** [reply]
Thanks for sharing this info! I'm reaching out to the Next.js engineers to see if they know more about this

**Reply by amyegan** [reply]
Hi again @ericmerl! I want to get some more info to clarify what is going wrong. Is the 404 affecting navigation for site visitors? Or is it only appearing in the logs?

If it is only in the logs, that is expected behavior when a new build is deployed. Next handles it automatically when a navigation occurs. But, if it's affecting navigation, then that is a problem we need to know more about.

**Reply by ericmerl** [reply]
Hey Amy, I did some more work on this and found that it was related to a client side error monitoring solution that I had installed (TrackJS). 

I didn't dig into it enough to know what the exact issue was, but it was likely just a mistake that I had made in the installation of the error monitoring. When I removed the monitoring code, the issue went away.

To answer your questions though, the 404 was affecting navigation for site visitors. Rather than reloading the page, the client side transition would happen with an empty object for the props, which was then caught by an error boundary. I'm wondering if the error monitoring I had incorrectly set up was somehow preventing next/link from catching that the error had occurred, causing it to handle the transition as if no errors had taken place.

**Reply by amyegan** [reply]
I'm glad you got it working!
